proto/proto-backplane-grpc/src/main/proto/deephaven/proto/table.proto

```diff
@@ -579,6 +594,222 @@ message ComboAggregateRequest {
   }
 }
 
+message AggregateAllRequest {
+  Ticket result_id = 1;
+  TableReference source_id = 2;
+  AggSpec spec = 3;
+  repeated string group_by_columns = 4;
+}
+
+message AggSpec {
+  message AggSpecApproximatePercentile {
+    // Percentile. Must be in range [0.0, 1.0].
+    double percentile = 1;
+
+    // T-Digest compression factor. Must be greater than or equal to 1. 1000 is extremely large.
+    // When not specified, the server will choose a compression value.
+    optional double compression = 2;
+  }
+
+  message AggSpecCountDistinct {
+    // Whether null input values should be included when counting the distinct input values.
+    bool count_nulls = 1;
+  }
+
+  message AggSpecDistinct {
+    // Whether null input values should be included in the distinct output values.
+    bool include_nulls = 1;
+  }
+
+  message AggSpecFormula {
+    // The formula to use to calculate output values from grouped input values.
+    string formula = 1;
+
+    // The formula parameter token to be replaced with the input column name for evaluation.
+    string param_token = 2;
+  }
+
+  message AggSpecMedian {
+    // Whether to average the highest low-bucket value and lowest high-bucket value, when the low-bucket and high-bucket
+    // are of equal size. Only applies to numeric types.
+    bool average_evenly_divided = 1;
+  }
+
+  message AggSpecPercentile {
+    // The percentile to calculate. Must be in the range [0.0, 1.0].
+    double percentile = 1;
+
+    // Whether to average the highest low-bucket value and lowest high-bucket value, when the low-bucket and high-bucket
+    // are of equal size. Only applies to numeric types.
+    bool average_evenly_divided = 2;
+  }
+
+  message AggSpecSorted {
+    // Using a message instead of string to support backwards-compatibility in the future
+    repeated AggSpecSortedColumn columns = 1;
+  }
+
+  message AggSpecSortedColumn {
+    // TODO(deephaven-core#821): SortedFirst / SortedLast aggregations with sort direction
+    string column_name = 1;
+  }
+
+  message AggSpecTDigest {
+    // T-Digest compression factor. Must be greater than or equal to 1. 1000 is extremely large.
+    // When not specified, the server will choose a compression value.
+    optional double compression = 1;
+  }
+
+  message AggSpecUnique {
+    // Whether to include null values as a distinct value for determining if there is only one unique value to output
+    bool include_nulls = 1;
+
+    // The output value to use for groups that don't have a single unique input value
+    AggSpecNonUniqueSentinel non_unique_sentinel = 2;
+  }
+
+  message AggSpecNonUniqueSentinel {
+    oneof type {
+      google.protobuf.NullValue null_value = 1;
+      string string_value = 2;
+      sint32 int_value = 3;
+      sint64 long_value = 4 [jstype=JS_STRING];
+      float float_value = 5;
+      double double_value = 6;
+      bool bool_value = 7;
+      // Expected to be in range [Byte.MIN_VALUE, Byte.MAX_VALUE]
```

Comment:
Something we could debate. Should we throw an error if the client tries to send a value that represents one of our null primitive values? (for int, long, float, double, etc) @rcaudy

Reply:
From a typing perspective, `null_value` doesn't have any type information associated with it, where-as `byte_value = Byte.MIN_VALUE` *does*. I'm not sure if that's a useful distinction we want to support or make w/ this structure.

Reply:
I might argue that errors wrt "null" primitive values should be the responsibility of the engine impl - if the engine allows explicitly setting `Integer.MIN_VALUE` as the sentinel value, we shouldn't try to stop that over the wire IMO.

Reply:
I'm ambivalent, here. I think it could be expedient to just accept the null values as in range, and assume the user wanted that, but I won't complain if we want to validate it and require them to use the `null` type.

Reply:
If we want to validate non-null primitives, we should either validate it in `io.deephaven.api.agg.spec.AggSpecUnique` or `io.deephaven.engine.table.impl.by.AggregationProcessor#makeUniqueOperator` IMO.

Reply:
And actually, it's more nuanced and should _probably_ be handled by `AggregationProcessor`. The implementation works right now be converting types based on the column type it's associated with based on `Number`. We don't have enough information at wire time, or `AggSpecUnique` time, to known whether the user is just trying to "save" bytes over the wire and take advantage of the implementations implicit coercions.